### PR TITLE
dont try to update storage mtime if we can't get the mtime

### PR DESCRIPTION
**Pull request by icewind1991**
same check as is already in place for `correctParentStorageMtime`

### CODE REVIEW

File: lib/private/Files/Cache/Updater.php [hunk 1]
```diff
@@ -224,12 +224,15 @@ public function renameFromStorage(IStorage $sourceStorage, $source, $target) {
...
-					'mtime' => null, // this magic tells it to not overwrite mtime
-					'storage_mtime' => $this->storage->filemtime($internalPath)
-				]
-			);
+			$mtime = $this->storage->filemtime($internalPath);
+			if ($mtime !== false) {
+				$this->cache->update(
+					$fileId, [
+						'mtime' => null, // this magic tells it to not overwrite mtime
+						'storage_mtime' => $this->storage->filemtime($internalPath)
```
Comment (rullzer) [inline lib/private/Files/Cache/Updater.php]: why not just use `$mtime`?

Comment (icewind1991) [reply lib/private/Files/Cache/Updater.php]: :see_no_evil:

**Review by rullzer — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**